apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/BooleanExpression.kt

```diff
@@ -109,7 +109,7 @@ fun <T : Any> BooleanExpression<T>.evaluate(block: (T) -> Boolean): Boolean {
   }
 }
 
-fun BooleanExpression<BTerm>.evaluate(variables: Set<String>, typename: String): Boolean {
+fun BooleanExpression<BTerm>.evaluate(variables: Set<String>, typename: String?): Boolean {
```

Comment:
The `typename` argument becomes nullable because we might want to call it in cases where we don't have a typename. For an example:

```graphql
query GetAnimal($withDetails: Boolean!) {
  animal {
    ... on Animal @include(if: $withDetails) {
      species
    }
  }
}
```

The code to evaluate the condition will be like:

```kotlin
variable("withDetails").evaluate(customScalarAdapters.variables(), null)
```